lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -55,21 +55,18 @@ internal sealed class ContextRuntime
         /// <summary>
         /// Create a new ContextRuntime.
         /// </summary>
+        /// <param name="id">ID of the context</param>
         /// <param name="serviceInjector"></param>
         /// <param name="contextConfiguration">the Configuration for this context.</param>
         /// <param name="parentContext"></param>
+        [Obsolete("Deprecated in 0.14, will be removed.")]
         public ContextRuntime(
+                string id,
                 IInjector serviceInjector,
                 IConfiguration contextConfiguration,
                 Optional<ContextRuntime> parentContext)
         {
-            var config = contextConfiguration as ContextConfiguration;
-            if (config == null)
-            {
-                Utilities.Diagnostics.Exceptions.Throw(
-                    new ArgumentException("contextConfiguration is not of type ContextConfiguration"), LOGGER);
-            }
-            _contextLifeCycle = new ContextLifeCycle(config.Id);
+            _contextLifeCycle = new ContextLifeCycle(id);
```

Comment:
We are having two paths, one is via injection, and one is via the to-be-removed path. This is the to-be-removed path.